pkg/client/sync_client.go

```diff
@@ -219,6 +220,47 @@ func (client *SyncClient) DownloadFromURL(downloadURL, filePath, uuid, diskUUID,
 	return nil
 }
 
+func (client *SyncClient) RestoreFromBackupURL(backupURL, concurrentLimit, filePath, uuid, diskUUID, expectedChecksum string, credential map[string]string) error {
+	httpClient := &http.Client{Timeout: 0}
+	encodedCredential, err := json.Marshal(credential)
+	if err != nil {
+		return err
+	}
+
+	requestURL := fmt.Sprintf("http://%s/v1/files", client.Remote)
+	req, err := http.NewRequest("POST", requestURL, bytes.NewReader(encodedCredential))
```

Comment:
credentials is a map type, we can only put it into body
in sync file server it will decode it back from the body